v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandlerTest.java

```diff
@@ -0,0 +1,981 @@
+  }
+
+  @Test
+  public void testHandleEmptyStringifiedJsonArray() {
+    String newValuesString = "[]";
+    JSONObject newValuesJson = new JSONObject();
+    newValuesJson.put("data", newValuesString);
+    String colKey = "data";
+    Set<Object> expected = new HashSet<>();
+    assertEquals(expected, handleStringifiedJsonToSet(colKey, newValuesJson));
+  }
+
+  @Test
+  public void testHandleNonArrayValue() {
+    String newValuesString = "\"apple\"";
+    JSONObject newValuesJson = new JSONObject();
+    newValuesJson.put("data", newValuesString);
+    String colKey = "data";
+    assertThrows(IllegalArgumentException.class, () -> handleStringifiedJsonToSet(colKey, newValuesJson));
+  }
+
+  @Test
+  public void testConvertToSmallIntValidInput() {
+    Integer validValue = 100;
+    short result = convertToSmallInt(validValue);
+    assertEquals(100, result);
+  }
+
+  @Test
+  public void testConvertToSmallIntBelowMinValue() {
+    Integer invalidValue = Short.MIN_VALUE - 1;
+    assertThrows(IllegalArgumentException.class, () -> convertToSmallInt(invalidValue));
+  }
+
+  @Test
+  public void testConvertToSmallIntAboveMaxValue() {
+    Integer invalidValue = Short.MAX_VALUE + 1;
+    assertThrows(IllegalArgumentException.class, () -> convertToSmallInt(invalidValue));
+  }
+
+  @Test
+  public void testConvertToTinyIntValidInput() {
+    Integer validValue = 100;
+    byte result = convertToTinyInt(validValue);
+    assertEquals(100, result);
+  }
+
+  @Test
+  public void testConvertToTinyIntBelowMinValue() {
+    Integer invalidValue = Byte.MIN_VALUE - 1;
+    assertThrows(IllegalArgumentException.class, () -> convertToTinyInt(invalidValue));
+  }
+
+  @Test
+  public void testConvertToTinyIntAboveMaxValue() {
+    Integer invalidValue = Byte.MAX_VALUE + 1;
+    assertThrows(IllegalArgumentException.class, () -> convertToTinyInt(invalidValue));
+  }
+
+  @Test
+  public void testEscapeCassandraStringNoQuotes() {
+    String input = "Hello World";
+    String expected = "Hello World";
+    String result = escapeCassandraString(input);
+    assertEquals(expected, result);
+  }
+
+  @Test
+  public void testEscapeCassandraStringWithSingleQuote() {
+    String input = "O'Reilly";
+    String expected = "O''Reilly";
+    String result = escapeCassandraString(input);
+    assertEquals(expected, result);
+  }
+
+  @Test
+  public void testEscapeCassandraStringEmpty() {
+    String input = "";
+    String expected = "";
+    String result = escapeCassandraString(input);
+    assertEquals(expected, result);
+  }
+
+  @Test
+  public void testEscapeCassandraStringWithMultipleQuotes() {
+    String input = "It's John's book.";
+    String expected = "It''s John''s book.";
+    String result = escapeCassandraString(input);
+    assertEquals(expected, result);
+  }
+
+  @Test
+  public void testConvertToCassandraTimestampWithValidOffset() {
+    String value = "2024-12-12T10:15:30+02:00";
+    String timezoneOffset = "+00:00";
```

Comment:
Test with a non-zero offset

Reply:
@shreyakhajanchi in cassandra we do not have offset I guess

Reply:
In that case why do you have this method:
`public static String convertToCassandraTimestamp(String value, String timezoneOffset) `

Reply:
Agree @shreyakhajanchi we are removing extra method in DML PR which is not required as of now we have added all posibility